avatars/avatars.js

```diff
@@ -408,6 +409,9 @@ class Avatar {
       };
     }
 
+    const stateName = "playerAvatar"
+    StateMachine.registerObj(options.name ?? stateName, this);
```

Comment:
I know this isn't used yet, but we cannot depend on `options.name` being unique, so this will collide and overwrite.

What was the intent here?

Reply:
Also, why is this mutating its argument rather than returning a tracker?

Reply:
Registering objects first checks to see if it's already being tracked on line 55 of StateMachine.js.  If it is, it won't reregister it, it will just take the object being passed and provide it with the tracker.